### DISCUSSION TITLE
How to run axisymmetric heat conduction simulation

### DISCUSSION THREAD

**Discussion by richmondodufisan** [idaholab/moose#28803] [Q&A Modules: General] [answered]
Hi, I am trying to run a simple heat conduction problem where I reduce the full 3D model to an axisymmetric 2D one. I am sure I'm missing something but I'm not sure what. I thought I'd ask here first before I get into the weeds of radial coordinates/working it out manually.

The problem setup is simple: a semi-infinite box with homogeneous material properties (thermal conductivity) has a gaussian heat flux applied at the top surface. It is steady state too, so no worry about the time terms for now.

![image_1](https://github.com/user-attachments/assets/a22d738d-ff58-46f8-acaa-4e471fde0eab)

Below is the temperature profile from the top of the box to the bottom at r = 0 (i.e x=0, y=0).

![graph_1](https://github.com/user-attachments/assets/60ac118a-9ce4-4f72-8baa-26a07f26f384)

I implemented the exact same simulation in 2D with a mesh that should be a slice of the 3D version. But my result differs. Not just in magnitude but the temperature profile itself is different. edit, it's based on the way derivatives are calculated in the rz coordinate system.

![image_2](https://github.com/user-attachments/assets/3a24030e-c52a-4960-adec-cf008dc3f336)

![graph_2](https://github.com/user-attachments/assets/ffb09bb1-a4b8-47fc-853c-5cc3b2a97701)

Is there an existing model/kernel that can help solve a problem like this/is there something simple I'm missing?

**Comment by richmondodufisan** [comment, marked as answer]
Solved, MOOSE has built in functionality, no need to reinvent the wheel

[Problem]
  coord_type = RZ
[]

Although I got a warning that this is deprecated and should be under the mesh block, but I got an error when I did that

**Reply by GiudGiud** [reply]
this should work
```
[Mesh]
  coord_type = RZ
  [other_mg]
      ...
  []
[]

**Reply by richmondodufisan** [reply]
That worked thanks!
